### PR TITLE
Include defmethod dispatch key in imenu items

### PR DESCRIPTION
**Pull request by vspinu**
All`(defmethod foo dispatch-key ...)` currently show as `foo` in imenu. Fix that by including corresponding `dispatch-key` into imenu items.

I have also replaced cl-destructuring-bind with let as it doesn't work with edebug in emacs 25.2.2. Cannot dig into that at the moment.

### CODE REVIEW

File: clojure-mode.el [hunk 1]
```diff
@@ -657,23 +657,30 @@ and `(match-end 1)'."
...
             (forward-sexp))
-          (or (if (char-equal ?[ (char-after (point)))
-                              (backward-sexp))
-                  (if (char-equal ?) (char-after (point)))
+          (or (when (char-equal ?[ (char-after (point)))
+                (backward-sexp))
+              (when (char-equal ?) (char-after (point)))
                 (backward-sexp)))
-          (cl-destructuring-bind (def-beg . def-end) (bounds-of-thing-at-point 'sexp)
+          (let* ((val (bounds-of-thing-at-point 'sexp))
```
Comment (bbatsov) [inline clojure-mode.el]: This change should ideally be a separate commit.

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@
 
 ### New features
 
+* Imenu understands defmethods
```
Comment (bbatsov) [inline CHANGELOG.md]: Added imenu support for multimethods.

**Comment by vspinu** [comment]
Fixed.

**Comment by vspinu** [comment]
I am adding one more commit to make imenu recognize indented def-forms. It's not uncommon to define  within a let block or a comment.
